caniusepython3/__main__.py

```diff
@@ -43,7 +43,10 @@ def projects_from_requirements(requirements):
             elif req.editable:
                 log.warning(
                     'Skipping {0}: editable projects unsupported'.format(req.name))
-            elif req.url and req.url.startswith('file:'):
+            elif getattr(req, 'url', None) and req.url.startswith('file:'):
```

Comment:
Since you are tossing the result of getattr(..., None) you should simply make these a hasattr check.